# HTTP Request/Response Cycle

[Open Notify](http://open-notify.org/)  is an an open source project to provide a simple programming interface for some of NASA’s awesome data. This takes live raw data from NASA's systems and turn them into APIs related to space and spacecraft. We can access the following information from open notify. 

* Current Location of the International Space Station

* Number of People in Space

* Overhead Pass Predictions for the International Space Station
    

### Current location of International Space Station

The first endpoint we'll look at on Open Notify is the ` iss-now.json` endpoint (current location of international space station). This endpoint gets the current latitude and longitude of the International Space Station.  Perform the following tasks 
* Make a get request to get the latest position of the international space station from the opennotify api's `iss-now` endpoint at http://api.open-notify.org/iss-now.json
* Check the status code of the response
* Interpret the returned code

In [2]:
import requests
iss_location = requests.get('http://api.open-notify.org/iss-now.json')

In [12]:
iss_location.status_code == requests.codes.ok

print(iss_location.status_code)

200


* Code 200 -- everything went okay, and the result has been returned

In [5]:
print(iss_location.text)

{"timestamp": 1692175870, "iss_position": {"latitude": "0.4831", "longitude": "112.7059"}, "message": "success"}


* The contents of the response are: {"timestamp", "iss_position", "message"}
* Its current location is: latitude": "0.4831", "longitude": "112.7059


### Number of people in space

In [17]:
# people = requests.get('http://api.open-notify.org/astros.json')

response =  requests.get('http://api.open-notify.org/astros.json')
data = response.json()
print(response.text)

{"number": 10, "people": [{"name": "Sergey Prokopyev", "craft": "ISS"}, {"name": "Dmitry Petelin", "craft": "ISS"}, {"name": "Frank Rubio", "craft": "ISS"}, {"name": "Stephen Bowen", "craft": "ISS"}, {"name": "Warren Hoburg", "craft": "ISS"}, {"name": "Sultan Alneyadi", "craft": "ISS"}, {"name": "Andrey Fedyaev", "craft": "ISS"}, {"name": "Jing Haiping", "craft": "Tiangong"}, {"name": "Gui Haichow", "craft": "Tiangong"}, {"name": "Zhu Yangzhu", "craft": "Tiangong"}], "message": "success"}


In [20]:
# number of people in space
print(data["number"])

# print(dict(data))

print(data['people'])



10
[{'name': 'Sergey Prokopyev', 'craft': 'ISS'}, {'name': 'Dmitry Petelin', 'craft': 'ISS'}, {'name': 'Frank Rubio', 'craft': 'ISS'}, {'name': 'Stephen Bowen', 'craft': 'ISS'}, {'name': 'Warren Hoburg', 'craft': 'ISS'}, {'name': 'Sultan Alneyadi', 'craft': 'ISS'}, {'name': 'Andrey Fedyaev', 'craft': 'ISS'}, {'name': 'Jing Haiping', 'craft': 'Tiangong'}, {'name': 'Gui Haichow', 'craft': 'Tiangong'}, {'name': 'Zhu Yangzhu', 'craft': 'Tiangong'}]
